Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,674 @@
+            ILogger<PluginManager> logger,
+            IApplicationHost appHost,
+            ServerConfiguration config,
+            string pluginsPath,
+            Version appVersion)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _pluginsPath = pluginsPath;
+            _appVersion = appVersion ?? throw new ArgumentNullException(nameof(appVersion));
+            _jsonOptions = JsonDefaults.GetOptions();
+            _config = config;
+            _appHost = appHost;
+            _nextVersion = new Version(_appVersion.Major, _appVersion.Minor + 2, _appVersion.Build, _appVersion.Revision);
+            _minimumVersion = new Version(0, 0, 0, 1);
+            _plugins = Directory.Exists(_pluginsPath) ? DiscoverPlugins().ToList() : new List<LocalPlugin>();
+        }
+
+        /// <summary>
+        /// Gets the Plugins.
+        /// </summary>
+        public IList<LocalPlugin> Plugins => _plugins;
+
+        /// <summary>
+        /// Returns all the assemblies.
+        /// </summary>
+        /// <returns>An IEnumerable{Assembly}.</returns>
+        public IEnumerable<Assembly> LoadAssemblies()
+        {
+            // Attempt to remove any deleted plugins and change any successors to be active.
+            for (int a = _plugins.Count - 1; a >= 0; a--)
+            {
+                var plugin = _plugins[a];
+                if (plugin.Manifest.Status == PluginStatus.Deleted && DeletePlugin(plugin))
+                {
+                    UpdateSuccessors(plugin);
+                }
+            }
+
+            // Now load the assemblies..
+            foreach (var plugin in _plugins)
+            {
+                CheckIfStillSuperceded(plugin);
+
+                if (plugin.IsEnabledAndSupported == false)
+                {
+                    _logger.LogInformation("Skipping disabled plugin {Version} of {Name} ", plugin.Version, plugin.Name);
+                    continue;
+                }
+
+                foreach (var file in plugin.DllFiles)
+                {
+                    Assembly assembly;
+                    try
+                    {
+                        assembly = Assembly.LoadFrom(file);
+                    }
+                    catch (FileLoadException ex)
+                    {
+                        _logger.LogError(ex, "Failed to load assembly {Path}. Disabling plugin.", file);
+                        ChangePluginState(plugin, PluginStatus.Malfunctioned);
+                        continue;
+                    }
+
+                    _logger.LogInformation("Loaded assembly {Assembly} from {Path}", assembly.FullName, file);
+                    yield return assembly;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates all the plugin instances.
+        /// </summary>
+        public void CreatePlugins()
+        {
+            var createdPlugins = _appHost.GetExports<IPlugin>(CreatePluginInstance)
```

Comment:
```suggestion
            _ = _appHost.GetExports<IPlugin>(CreatePluginInstance)
```

Unused variable, can be discarded